src/osd/ReplicatedPG.cc

```diff
@@ -9760,9 +9750,14 @@ void ReplicatedPG::_applied_recovered_object(ObjectContextRef obc)
   --active_pushes;
 
   // requeue an active chunky scrub waiting on recovery ops
-  if (!deleting && active_pushes == 0
+  if (!deleting && active_pushes == 0 &&
```

Comment:
Ah, I guess you did update these users.  Still, active_pushes could continue to be 0, right?  If do_scrub_map resets active_rep_scrub again before requeueing the other event, wouldn't this refire?